### PR TITLE
Display hardware costs in the transaction list and details page

### LINKED ISSUE #1138: Display hardware costs for the period in the transaction list with link to the details page
i.e. the cost of a card reader and shipping.

paJDYF-Xe-p2

### Description

Transactions in the transaction list may contain a `metadata` section with additional detals.

If `metadata` is similar to the following example, it indicates a hardware charge:
```json
  "metadata": {
        "interval_from": "2021-08-01",
        "interval_to": "2021-08-31",
        "charge_type": "card_reader_fee"
      }
```

In this case clicking on the transaction line should take merchant to the details page (see reference below).

### References

 * paJDYF-2Bw-p2
 * p1633533995293900-slack-CGGCLBN58

### PR DESCRIPTION
**Pull request by dpaun1985**
Fixes #1138 

#### Changes proposed in this Pull Request

This PR updates the transactions list to include card reader costs, and also updates the details page to display information about charges for active reader.

Changes:
- add new endpoint to fetch card reader costs
- update the transactions list to include rows with card reader costs. 
image (1)
- update details page to display information about charges for card readers. This screen will be shown when you click the card reader row from transactions list.
Screenshot 2021-10-22 at 14 06 56

#### Testing instructions

* Debit merchant account for active Card Readers used in a month 
* Go to transactions screen and check if the hardware costs are displayed as above
* Click on the hardware costs raw and check if the details are displayed correctly.

-------------------

- [ ] Added changelog entry (or does not apply)
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Added testing instructions to the [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) (or does not apply)

### CODE REVIEW

**Review by kalessil — COMMENTED** [review]
On a high level, the approach looks good and feels consistent with other client parts !

**Review by dmvrtx — COMMENTED** [review]
Overall looks good, great job! I left a couple comments in places which were not clear to me.

File: includes/admin/class-wc-rest-payments-reader-charges-controller.php [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	 */
+	protected $rest_base = 'payments/reader-charges';
+
+	/**
+	 * Configure REST API routes.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/summary',
```
[the remark this answers is not in the document]
Comment (dpaun1985) [reply includes/admin/class-wc-rest-payments-reader-charges-controller.php]: indeed. added the transaction id param.

**Comment by anu-rock** [comment]
@dpaun1985 The new changes LGTM! Thanks for taking care of 'reader' → 'card reader'. 

Question - Should the route for reader charges be `payments/readers/charges`? That would help implement the design as seen here HsXXAeasBmnvvYynYC29Je-fi-2%3A2810 without further refactoring changes in this PR later. Eg. to get a list of readers attached to an account - `GET payments/readers`. Feel free to ignore if it sounds like over-engineering.

**Comment by dpaun1985** [comment]
> @dpaun1985 The new changes LGTM! Thanks for taking care of 'reader' → 'card reader'.
> 
> Question - Should the route for reader charges be `payments/readers/charges`? That would help implement the design as seen here [HsXXAeasBmnvvYynYC29Je-fi-2%3A2810](https://href.li/?https://www.figma.com/file/HsXXAeasBmnvvYynYC29Je/?node-id=2%3A2810) without further refactoring changes in this PR later. Eg. to get a list of readers attached to an account - `GET payments/readers`. Feel free to ignore if it sounds like over-engineering.

Thank you for the remark. Indeed, the naming of the route should be changed. Just pushed the changes.

**Review by anu-rock — APPROVED** [review]
LGTM :shipit:

**Review by kalessil — CHANGES_REQUESTED** [review]
Yay, this PR is shaping really great!

2 notes:
- can we add a changelog entry, please?
- When rendering fee/amount I noticed we use 'N/A', is that by design?

EDIT: the last question is only half-actual: can we display Amount as 0.0 in the corresponding currency for transactions list, please (instead of N/A)?

**Comment by dpaun1985** [comment]
> Yay, this PR is shaping really great!
> 
> 2 notes:
> 
> * can we add a changelog entry, please?
> * When rendering fee/amount I noticed we use 'N/A', is that by design?
> 
> EDIT: the last question is only half-actual: can we display Amount as 0.0 in the corresponding currency for transactions list, please (instead of N/A)?

Thank you.  Pushed the changes.

**Review by kalessil — APPROVED** [review]
LGTM, thanks for the amount tweak!

**Pull request merged**